File: 0003-longest-substring-without-repeating-characters/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {

        int[] has = new int[256];
        Arrays.fill(has,-1);
        int check = -1 , res = 0 ;

        int n = s.length();
        for(int i = 0 ; i < n ; i++){
            if(has[s.charAt(i)] > check){
                check = has[s.charAt(i)];
            }
            res = Math.max(res , i - check);
            has[s.charAt(i)] = i ;
        }
        
        return res ;
    }
}

File: 0003-longest-substring-without-repeating-characters/solution.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        const int n = s.size();
        vector<int> v(256,-1);
        int res = 0 , ck = -1 ;
        for(int i = 0 ; i < n ; i++ ){
                if(v[s[i]] > ck)
                    ck = v[s[i]];
                v[s[i]] = i ;
                res = max(res,i-ck);
        }
        
        return res ;
    }
};

